src/util/mod.rs

```diff
@@ -146,25 +145,35 @@ pub fn to_socket_addr<A: ToSocketAddrs>(addr: A) -> io::Result<SocketAddr> {
     Ok(addrs.collect::<Vec<SocketAddr>>()[0])
 }
 
-/// A struct to make byte array format more friendly.
+/// A function to escape a byte array to a readable ascii string.
 ///
-/// Remove this once issue [33127](https://github.com/rust-lang/rust/issues/33127) is resolved.
-pub struct HexDisplay<'a> {
-    data: &'a [u8],
-}
-
-impl<'a> Display for HexDisplay<'a> {
-    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
-        try!(write!(f, "0x"));
-        for &b in self.data {
-            try!(write!(f, "{:02X}", b));
+/// # Examples
+///
+/// ```
+/// use tikv::util::escape;
+///
+/// assert_eq!("ab", escape(b"ab"));
+/// assert_eq!("a\\015\\012\\011 \\'\\\"\\", escape(b"a\r\n\t '\"\\"));
+/// assert_eq!("\\342\\235\\244\\360\\237\\220\\267", escape("❤🐷".as_bytes()));
+/// ```
+///
+pub fn escape(data: &[u8]) -> String {
+    let mut escaped = Vec::with_capacity(data.len() * 4);
```

Comment:
Former implementation seems better since it writes to the `Formatter` directly and avoids memory allocation.